app/models/ability.rb

```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the user here. For example:
+    return unless user.present?
+
+    can :manage, Comment, author_id: user.id
+    can :manage, Post, author_id: user.id
+    can :read, :all
+    return unless user.role == 'admin'
+
+    can :manage, :all
```

Comment:
- [ ] Great work on making the authorization for normal users 👍 but it would be nice if you make a way to register as an admin in your application because I can test everything as a normal user I can test your authorization but i cannot check authorization as a admin because 😨 Kindly make a way so a user can be an admin 👍

Reply:
since we are running short of time we also have fourth project to complete. Kindly mark this requirement as optional.

Reply:
**[REQUESTED CHANGE]**

- Hey @baqar-abbas, while you've not implemented the changes requested by the previous reviewer, it's mandatory that you add the requested changes as it's a major part of the project requirement. I'd try to explain a little. Please, kindly refer to the [project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_add_authorization.md) to see it stated clearly.

![image](https://github.com/Paul-tes/Blog-App/assets/59397159/889b1716-30c3-4b83-bfb6-21d66fab3731)


> In the case of the requested change according to the project requirement,  A user can only and should only be able to delete their own comments and posts, and not other comments and posts by another user, in this case, only the admin can delete any posts or comments of any user. This is why you should add the role of the admin to be able to meet this part of the project requirement.